test/api/attributes/[attributeId].test.ts

```diff
@@ -0,0 +1,171 @@
+import * as auth from 'utils/auth'
+import * as MongoDriver from 'server/actions/MongoDriver'
+import * as apiValidator from 'utils/apiValidators'
+import mongoose from 'mongoose'
+import { clientPromise } from '@api/auth/[...nextauth]'
+
+beforeAll(() => {
+  jest.spyOn(auth, 'serverAuth').mockImplementation(() => Promise.resolve())
+
+  jest.spyOn(apiValidator, 'apiObjectIdValidation').mockImplementation()
+})
+
+// restore mocked implementations and close db connections
+afterAll(() => {
+  jest.restoreAllMocks()
+  mongoose.connection.close()
+  clientPromise.then((client) => client.close())
+})
+
+describe('api/attributes/[attributeId]', () => {
+  test('thrown error is caught, response is unsuccessful and shows correct error message', async () => {
+    const mockObjectId = '6408a7156668c5655c25b105'
+
+    jest.spyOn(auth, 'serverAuth').mockImplementationOnce(async () => {
+      throw new ApiError(401, 'Unauthorized')
+    })
+
+    const request = createRequest({
+      method: 'GET',
+      url: `/api/attributes/${mockObjectId}`,
+      query: {
+        attributeId: mockObjectId,
+      },
+    })
+    const response = createResponse()
+
+    await handler(request, response)
+
+    const data = response._getJSONData()
+
+    expect(response.statusCode).toBe(401)
+    expect(data.message).toBe('Unauthorized')
+    expect(data.success).toBe(false)
+  })
+
+  test('unsupported method returns 405', async () => {
+    const mockObjectId = '6408a7156668c5655c25b105'
+
+    const request = createRequest({
+      method: 'POST',
+      url: `/api/attributes/${mockObjectId}`,
+      query: {
+        attributeId: mockObjectId,
+      },
+    })
+    const response = createResponse()
+
+    await handler(request, response)
+
+    const data = response._getJSONData()
+
+    expect(response.statusCode).toBe(405)
+    expect(data.message).toBe('Method Not Allowed')
+    expect(data.success).toBe(false)
+  })
+  describe('GET', () => {
+    test('valid call returns correct data', async () => {
+      const mockObjectId = '6408a7156668c5655c25b105'
+
+      const mockGetEntity = jest
+        .spyOn(MongoDriver, 'getEntity')
+        .mockImplementation(
+          async () =>
+            validAttributeResponse as AttributeDocument & { _id: ObjectId }
+        )
+
+      const request = createRequest({
+        method: 'GET',
+        url: `/api/attributes/${mockObjectId}`,
+        query: {
+          attributeId: mockObjectId,
+        },
+      })
+
+      const response = createResponse()
+
+      await handler(request, response)
+      const data = response._getJSONData().payload
+
+      expect(mockGetEntity).toHaveBeenCalledTimes(1)
+      expect(mockGetEntity).lastCalledWith(AttributeSchema, mockObjectId)
+      expect(response.statusCode).toBe(200)
+      expect(data).toEqual(validAttributeResponse)
+    })
+  })
+
+  describe('PUT', () => {
+    jest.spyOn(apiValidator, 'apiAttributeValidation').mockImplementation()
```

Comment:
I believe this can be removed, as it's a duplicate of another mock inside the test

Reply:
not sure what you mean here, the whole test can be deleted or this line?

Reply:
Just this one line. the same `mockImplementation` line appears on line 111.

Reply:
removed